### PR TITLE
STORE-1198 fix, Implement new method to check permission for an asset given the registry path

### PR DESCRIPTION
**Pull request by tmkasun**

### CODE REVIEW

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js [hunk 1]
```diff
@@ -946,6 +946,36 @@ var permissions = {};
...
+
+        if ((!tenantId)) {
+            throw 'Unable to resolve permissions without the tenantId';
+        }
+        var userManager = server.userManager(tenantId);
+        var authorizer = userManager.authorizer;
+
+        try {
+            if ((!username) || (username === wso2AnonUsername())) {
+                if(log.isDebugEnabled()){
```
Comment (daneshk) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: formatting issue

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js [hunk 2]
```diff
@@ -946,6 +946,36 @@ var permissions = {};
         }
         return authorized;
     };
+
+    permissions.SYSTEM_ANON_ROLE = 'SYSTEM/wso2.anonymous.role';
+    permissions.checkActionPermissionforPath = function(path,action,tenantId,username) {
```
Comment (mushthaq33) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: Formatting issue

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js [hunk 3]
```diff
@@ -946,6 +946,36 @@ var permissions = {};
...
+        var authorizer = userManager.authorizer;
+
+        try {
+            if ((!username) || (username === wso2AnonUsername())) {
+                if(log.isDebugEnabled()){
+                    log.debug('[permissions] username not provided to check ' + permission + '.The anon role will be used to check permissions');
+                }
+                isAuthorized = authorizer.isRoleAuthorized(permissions.SYSTEM_ANON_ROLE, path, action);
+            } else {
+                if(log.isDebugEnabled()) {
```
Comment (mushthaq33) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: Formatting issue

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js (hunk 2 above)
Comment (tmkasun) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/e95caa53ae6d2bec4d278537d014af0496cdfa7a)

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js (hunk 3 above)
Comment (tmkasun) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/e95caa53ae6d2bec4d278537d014af0496cdfa7a)

File: jaggery-modules/rxt/module/scripts/permissions/permissions.js (hunk 1 above)
Comment (tmkasun) [inline jaggery-modules/rxt/module/scripts/permissions/permissions.js]: Thanks @daneshk  , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/e95caa53ae6d2bec4d278537d014af0496cdfa7a)

**Pull request merged**